deploy/deploy.lib

```diff
@@ -173,6 +173,27 @@ list_module_paths() {
     done
 }
 
+list_module_archive_paths() {
+    # Print all module paths up to and including the stage passed as first
+    # argument. Module paths are taken from the archive location.
+    local what="$1"
+    for stage in ${stages}; do
+        local modddir="${DEPLOYMENT_ROOT}/modules/${what}"
+        if [[ -d "${moddir}" ]]; then
+            local base="$(find ${moddir} -maxdepth 1 -type d -regex '.*/[0-9]+-[0-9]+'|sort|tail -n 1)"
+            for modpath in ${base}/modules/tcl/*; do
+                if [[ -d "${modpath}" ]]; then
```

Comment:
probably missing why we need to explicitly check if dir exist

Reply:
There are files in that directory, and we don't want to grab them.